### DISCUSSION TITLE
[Pytorch CUDA] 如何使用xmake为Pytorch编译CUDA算子？

### DISCUSSION THREAD

**Discussion by shenmishajing** [xmake-io/xmake#3962] [Q&A]
### 你在什么场景下需要该功能？

已经有很多教程关于使用cmake为Pytorch编译C++或CUDA算子了，xmake可以实现相同的东西吗？

一个可能的CMakeLists.txt:
https://github.com/godweiyang/NN-CUDA-Example/blob/master/pytorch/CMakeLists.txt

### 描述可能的解决方案

一个能编译类似的Pytorch C++/CUDA 算子的xmake.lua，或者有可能的话，一个CMakeLists.txt转xmake.lua的工具？

[a comment by an automated account is not shown]

**Comment by waruqi** [comment]
没试过，你可以研究下

[a comment by an automated account is not shown]

**Comment by waruqi** [comment]
这个应该就是 用 libtorch 包 + cuda 编译，好像也没啥特别的。。

```lua
add_requires("libtorch")
target("test")
    set_kind("binary")
    add_files("src/*.cu")
    add_files("src/*.cpp")
    add_packages("libtorch")
```

**Reply by shenmishajing** [reply]
能用conda环境里安装的Pytorch吗？cmake那边是在conda环境里安装一个Pytorch然后把相关的路径添加到CMAKE_PREFIX_PATH里就可以了，这样就不用再单独安装一遍libtorch了。

**Reply by wangyexiang** [reply]
请问，你搞定了吗？@shenmishajing
